internal/formatter/formatter.go

```diff
@@ -0,0 +1,94 @@
+package formatter
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+)
+
+// Format method formats files inside of the integration directory.
+func Format(packageRoot string, failFast bool) error {
+	err := filepath.Walk(packageRoot, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if info.IsDir() {
+			return nil
+		}
+		err = formatFile(path, failFast)
+		if err != nil {
+			return errors.Wrapf(err, "formatting file failed (path: %s)", path)
+		}
+
+		return nil
+	})
+	if err != nil {
+		return errors.Wrap(err, "walking through the integration files failed")
+	}
+	return nil
+}
+
+func formatFile(path string, failFast bool) error {
+	file := filepath.Base(path)
+	ext := filepath.Ext(file)
+
+	switch ext {
+	case ".json":
+	case ".yml":
+	default:
+		return nil
+	}
+
+	content, err := ioutil.ReadFile(path)
+	if err != nil {
+		return errors.Wrap(err, "reading file content failed")
+	}
+
+	var formatted []byte
+	switch ext {
+	case ".json":
+		var rawMessage json.RawMessage
+		err = json.Unmarshal(content, &rawMessage)
+		if err != nil {
+			return errors.Wrap(err, "unmarshalling JSON file failed")
+		}
+
+		formatted, err = json.MarshalIndent(&rawMessage, "", "    ")
+		if err != nil {
+			return errors.Wrap(err, "marshalling JSON raw message failed")
+		}
+	case ".yml":
+		// yaml.Unmarshal() requires `yaml.Node` to be passed instead of generic `interface{}`.
+		// Otherwise it can detect any comments and fields are considered as normal map.
+		var node yaml.Node
+		err = yaml.Unmarshal(content, &node)
+		if err != nil {
+			return errors.Wrap(err, "unmarshalling YAML file failed")
+		}
+
+		formatted, err = yaml.Marshal(&node)
+		if err != nil {
+			return errors.Wrap(err, "marshalling JSON raw message failed")
+		}
+	}
```

Comment:
So we end up having two switch-case constructs in this function that are switching on the same cases. I don't think this is a big deal right now as we only have two cases so feel free to keep things as-is. 

However, just as a suggestion, if you like you could consider doing something like:

```go
type formatter func(path string) ([]byte, bool, error)

var formatters = map[string]formatter{
  "json": jsonFormatter,
  "yml": yamlFormatter,
  "yaml": yamlFormatter,
}

func jsonFormatter(filePath []byte) ([]byte, bool, error) {
   // return formatted bytes, whether file is already in good shape, error
}

// more formatter functions
```

and then use the map inside this function instead of the two `switch-case` constructs.

Reply:
I like this design idea! I've adjusted the code.